pkg/ruler/api_test.go

```diff
@@ -415,11 +419,35 @@ interval: 15s
 			err:    errors.New("invalid rules config: rule group 'rg_name' has no rules"),
 		},
 		{
-			name:   "with a a valid rules file",
+
+			name:   "with federated rules without enabled federation",
+			cfg:    defaultCfg,
+			status: 400,
+			input: `
+name: test
+interval: 15s
+source_tenants: [t1, t2]
+rules:
+- record: up_rule
+  expr: up{}
+- alert: up_alert
+  expr: sum(up{}) > 1
+  for: 30s
+  annotations:
+    test: test
+  labels:
+    test: test
+`,
+			err: errors.New("invalid rules config: rule group 'test' is a federated rule group but rules federation is disabled. To enable it, set -ruler.tenant-federation.enabled=true as a CLI argument or ruler.tenant_federation.enabled: true in YAML or contact your service administrator"),
```

Comment:
Is this message going to be shown by `mimirtool`? If so, I wonder if we need to be clearer that the CLI flag and YAML configuration need to be set on the server side - the message might be misinterpreted to mean that the CLI flag needs to be added to the `mimirtool` invocation or the rules YAML file.

Reply:
good call. I pushed a change with a reworded message, but I feel like I could use a quick review from @osg-grafana